src/core/tests/type_prop/strided_slice.cpp

```diff
@@ -651,3 +651,69 @@ INSTANTIATE_TEST_SUITE_P(type_prop,
                                  element::f32              // reference type
                              }),
                          StridedSliceShapeInferTest::get_test_case_name);
+
+using StridedSliceIntervalParams =
+    std::tuple<ov::PartialShape, ov::PartialShape, ov::PartialShape, int64_t, int64_t, int64_t, ov::PartialShape>;
+
+class StridedSliceIntervalTest : public TypePropOpTest<op::v1::StridedSlice>,
+                                 public WithParamInterface<StridedSliceIntervalParams> {
+protected:
+    void SetUp() override {
+        std::tie(data_shape, begin_shape, end_shape, begin_offset, end_offset, step, exp_shape) = GetParam();
+    }
+
+    ov::PartialShape data_shape, begin_shape, end_shape, exp_shape;
+    int64_t begin_offset, end_offset, step;
+};
+
+INSTANTIATE_TEST_SUITE_P(type_prop,
+                         StridedSliceIntervalTest,
+                         Values(StridedSliceIntervalParams({1024}, {{0, 20}}, {{10, 20}}, 0, 0, 1, {{0, 20}}),
+                                StridedSliceIntervalParams({1024}, {{0, 20}}, {{10, 20}}, 0, 0, 1, {{0, 20}}),
+                                StridedSliceIntervalParams({1024}, {{0, 20}}, {{10, 20}}, 10, 0, 1, {{0, 20}}),
+                                StridedSliceIntervalParams({-1}, {{0, 20}}, {{10, 20}}, 10, 0, 1, {{0, 20}}),
+                                StridedSliceIntervalParams({1024}, {{10, 20}}, {{0, 20}}, 0, 10, 1, {{0, 1013}}),
+                                StridedSliceIntervalParams({1024}, {{0, 10}}, {{0, 5}}, 0, 10, 1, {{1004, 1019}}),
+                                StridedSliceIntervalParams({{120, 1024}}, {{0, 10}}, {{0, 5}}, 0, 10, 1, {{100, 1019}}),
+                                StridedSliceIntervalParams({1024}, {{0, 1030}}, {{0, 2000}}, 1025, 10, 1, {{0, 1024}}),
+                                StridedSliceIntervalParams({1024}, {{0, 10}}, {{10, 20}}, 10, 0, 1, {{0, 20}}),
+                                StridedSliceIntervalParams({1024}, {{1, 12}}, {{0, 18}}, 10, 0, 2, {{0, 9}}),
+                                StridedSliceIntervalParams({1024}, {{10, 20}}, {{0, 20}}, 0, 10, 2, {{0, 507}}),
+                                StridedSliceIntervalParams({{100, 1024}}, {{10, 20}}, {{0, 20}}, 0, 10, 2, {{0, 507}}),
+                                StridedSliceIntervalParams({1024}, {10}, {30}, 0, 0, 2, {10}),
+                                StridedSliceIntervalParams({1024}, {{10, 15}}, {30}, 0, 0, 2, {{8, 10}}),
+                                StridedSliceIntervalParams({1024}, {{10, 15}}, {{30, 40}}, 0, 0, 2, {{8, 15}}),
+                                StridedSliceIntervalParams({{20, 1024}}, {{10, 15}}, {{30, 40}}, 0, 0, 2, {{3, 15}}),
```

Comment:
```suggestion
                                StridedSliceIntervalParams({{20, 1024}}, {{10, 15}}, {{30, 40}}, 0, 0, 2, {{3, 15}}),
                                StridedSliceIntervalParams({1024}, {{0, 1030}}, {{0, 2000}}, 1020, 10, 1, {{0, 1014}}),
```
![photo_5332764590695630072_y](https://user-images.githubusercontent.com/55839243/233054685-2e550db0-816f-4489-baad-76da797e1d33.jpg)

Please investigate this case too

Reply:
@jane-intel The example above doesn't follow numpy rules, please take a look:

```
>>> a = np.arange(1024)
>>> a
array([   0,    1,    2, ..., 1021, 1022, 1023])

1.
>>> a[-1020 : -10]
array([   4,    5,    6, ..., 1011, 1012, 1013])
>>> len(a[-1020 : -10])
1010

2.
>>> a[10 : -10]
array([  10,   11,   12, ..., 1011, 1012, 1013])
>>> len(a[10 : -10])
1004

3.
>>> a[-1020 : 1990]
array([   4,    5,    6, ..., 1021, 1022, 1023])
>>> len(a[-1020 : 1990])
1020

4.
>>> a[10 : 1990]
array([  10,   11,   12, ..., 1021, 1022, 1023])
>>> len(a[10 : 1990])
1014
```

But within begin values {-1020; 10} there can be also 0, 
so the maximum range is from begin 0 to end 1990:

```
>>> a[0 : 1990]
array([   0,    1,    2, ..., 1021, 1022, 1023])
>>> len(a[0 : 1990])
1024
```
What gives the final output shape interval {0, 1024}